content/en/guide/v10/signals.md

```diff
@@ -356,6 +356,16 @@ name.value = "John";
 // Logs: "John Doe"
 ```
 
+You can return a function from the function provided to [`effect()`](#effectfn) that will be run right before the next update takes place. The purpose of the returned function is to clean up the side effect before the next update.
```

Comment:
```suggestion
Optionally, you can return a cleanup function from the callback provided to [`effect()`](#effectfn) that will be run before the next update takes place. This allows you to "clean up" the side effect and potentially reset any state for the subsequent trigger of the callback.
```

I don't think this section fits where you've inserted it though, would rather have this example & description in the API docs below.

Reply:
As far as I understand the API docs are supposed to be reference for people who already know how a feature works. So people should be able to see a short explanation and example of cleanup functions before going to the reference.

Reply:
I don't really agree there, API docs are for both new users and familiar ones alike, but we can keep this I suppose.